paper

```diff
@@ -0,0 +1,298 @@
+        printf '                        | respectively to edit the correct project. Use the argument "continue" after\n'
+        printf '                        | the changes have been made to finish and rebuild patches. Can be called from anywhere.\n'
+        printf '\n'
+        printf '  * setup               | Add an alias to %s to allow full functionality of this script. Run as:\n' "${paper_rcpath}"
+        printf '                        |     . ./paper setup\n'
+        printf "                        | After you run this command you'll be able to just run \"paper\" from anywhere.\n"
+        printf '                        | The default name for the resulting alias is "paper", you can give an argument to override\n'
+        printf '                        | this default, such as:\n'
+        printf '                        |     . ./paper setup example\n'
+        printf '                        | Which will allow you to run "example" instead.\n'
+  }
+
+  ###########################
+  ### CHECK PREREQUISITES ###
+  ###########################
+  local -r java_not_found_message="was not found - download the JDK by following https://paper.readthedocs.io/en/latest/java-update/index.html"
+  # TODO: Verify which ones are required
+  _paper_is_dep_available git
+  _paper_is_dep_available patch
+  _paper_is_dep_available mvn
+  _paper_is_dep_available curl
+  _paper_is_dep_available javac "${java_not_found_message}"
+  _paper_is_dep_available jar "${java_not_found_message}"
+
+  ###############
+  ### LETS GO ###
+  ###############
+  case "${1:-}" in
+    rb|rbp|rebuild|rbf|rbfull)
+      (
+        local target_task
+        case "${2:-}" in
+          ''|all) target_task=rebuildPatches;;
+          a|api) target_task=rebuildApiPatches;;
+          s|server) target_task=rebuildServerPatches;;
+          *) printf 'Unknown target - %s\nExpected one of: all api server\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        if [[ "${1}" == @(rbf|rbfull) ]]; then
+          printf 'Rebuilding patches without filter\n' >&2
+          ./gradlew "${target_task}" --filter-patches=false || return "${?}"
+        else
+          ./gradlew "${target_task}" || return "${?}"
+        fi
+      );;
+    p|patch)
+      (
+        local target_task
+        case "${2:-}" in
+          ''|all) target_task=applyPatches;;
+          a|api) target_task=applyApiPatches;;
+          s|server) target_task=applyServerPatches;;
+          *) printf 'Unknown target - %s\nExpected one of: all api server\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        ./gradlew "${target_task}" || return "${?}"
+      );;
+    j|jar)
+      (
+        cd "${basedir}"
+        ./gradlew applyPatches || return "${?}"
+        ./gradlew paperclipJar || return "${?}"
+      );;
+    b|build)
+      (
+      printf 'NOT IMPLEMENTED YET\n' >&2 # TODO
```

Comment:
This shouldn't be any more complicated than `./gradlew build`

Reply:
Done.